Roles/AddOns/Common/Rebirth.cs

```diff
@@ -0,0 +1,61 @@
+﻿using static TOHE.Options;
+using static TOHE.Utils;
+using static TOHE.Translator;
+using TOHE.Modules;
+using static UnityEngine.GraphicsBuffer;
+using TOHE.Roles.Neutral;
+
+namespace TOHE.Roles.AddOns.Common;
+
+public class Rebirth : IAddon
+{
+    private const int Id = 29500;
+    public AddonTypes Type => AddonTypes.Helpful;
+    public static OptionItem RebirthUses;
+    public static Dictionary<byte, int> Rebirths = [];
+    public void SetupCustomOption()
+    {
+        SetupAdtRoleOptions(Id, CustomRoles.Rebirth, canSetNum: true, teamSpawnOptions: true);
+        RebirthUses = IntegerOptionItem.Create(Id + 11, "RebirthUses", new(1, 14, 1), 1, TabGroup.Addons, false).SetParent(Options.CustomRoleSpawnChances[CustomRoles.Rebirth])
+           .SetValueFormat(OptionFormat.Times);
+    }
+    public static void Add(byte Playerid)
+    {
+        Rebirths[Playerid] = RebirthUses.GetInt();
+    }
+    public static void Remove(byte Playerid) 
+    {
+        Rebirths.Remove(Playerid);
+    }
+
+    public static bool SwapSkins(PlayerControl pc, out NetworkedPlayerInfo NewExiledPlayer)
+    {
+        NewExiledPlayer = default;
+        if (!pc.Is(CustomRoles.Rebirth)) return false;
+        var ViablePlayer = Main.AllAlivePlayerControls.Where(x => x != pc).Shuffle(IRandom.Instance)
+            .FirstOrDefault(x => x != null && !x.OwnedByHost() && !x.IsAnySubRole(x => x.IsConverted()) && !x.Is(CustomRoles.Admired) && !x.Is(CustomRoles.Knighted) && 
+/*All converters */    (!x.Is(CustomRoles.Cultist) && !x.Is(CustomRoles.Infectious) && !x.Is(CustomRoles.Virus) && !x.Is(CustomRoles.Jackal) && !x.Is(CustomRoles.Admirer)) &&
+                !x.Is(CustomRoles.Lovers) && !x.Is(CustomRoles.Romantic) && !x.GetCustomRole().IsImpostor());
```

Comment:
I also don’t see the code that checks whether the player voted for you (I read your suggestion in discord)

Reply:
oh I completely forgot about that part 💀 gonna add this now